yt_dlp/extractor/generic.py

```diff
@@ -2373,6 +2374,14 @@ def _real_extract(self, url):
         else:
             video_id = self._generic_id(url)
 
+        # Try to impersonate a web-browser by default if possible
+        # We test if impersonation is available at all to omit the warning if it isn't
+        impersonate = None
+        if self._downloader._impersonate_target_available(ImpersonateTarget()):
+            targets = self._configuration_arg('impersonate', [''])
+            if targets != ['false']:
+                impersonate = list(map(ImpersonateTarget.from_str, targets))
```

Comment:
this kinda duplicates what we are already doing in `InfoExtractor._request_webpage`. we could simplify it like this:
```suggestion
        impersonate = self._configuration_arg('impersonate', [''])
        if 'false' in impersonate or not self._downloader._impersonate_target_available(ImpersonateTarget()):
            impersonate = None
```